manager/src/main/java/com/redhat/service/smartevents/manager/connectors/ConnectorsServiceImpl.java

```diff
@@ -54,21 +57,21 @@ public void createConnectorEntity(Processor processor) {
 
     @Transactional(Transactional.TxType.MANDATORY)
     private void createConnectorEntity(Processor processor, Action action) {
-        Optional<GatewayConnector<Action>> optActionConnector = gatewayConfigurator.getActionConnector(action.getType());
-        if (optActionConnector.isEmpty()) {
+        if (!processorCatalogService.isConnector(ProcessorType.SINK, action.getType())) {
             return;
         }
         String topicName = gatewayConfiguratorService.getConnectorTopicName(processor.getId());
-        GatewayConnector<Action> actionConnector = optActionConnector.get();
-        persistConnectorEntity(processor, topicName, actionConnector.getConnectorType(), actionConnector.getConnectorTypeId(), actionConnector.connectorPayload(action, topicName));
+        persistConnectorEntity(processor, topicName, ConnectorType.SINK, action.getType(), gatewayConnector.connectorPayload(action, topicName));
     }
 
     @Transactional(Transactional.TxType.MANDATORY)
     // Connector should always be marked for creation in the same transaction as a Processor
     public void createConnectorEntity(Processor processor, Source source) {
-        GatewayConnector<Source> sourceConnector = gatewayConfigurator.getSourceConnector(source.getType());
+        if (!processorCatalogService.isConnector(ProcessorType.SOURCE, source.getType())) {
+            return;
+        }
         String topicName = gatewayConfiguratorService.getConnectorTopicName(processor.getId());
-        persistConnectorEntity(processor, topicName, sourceConnector.getConnectorType(), sourceConnector.getConnectorTypeId(), sourceConnector.connectorPayload(source, topicName));
+        persistConnectorEntity(processor, topicName, ConnectorType.SOURCE, source.getType(), gatewayConnector.connectorPayload(source, topicName));
```

Comment:
as we take exactly the same parameters specified by managed connectors I implemented the logic in just one `GatewayConnector` -> we do not need anymore one class for every action/source